src/main/java/com/booleanuk/Poker.java

```diff
@@ -28,4 +107,12 @@ public String extensionThreeMethods() {
         // You will also need to add the relevant tests to Extension3Test.java
         return "Replace this method with your own methods and tests";
     }
+
+    public static void main(String[] args) {
+        Poker poker = new Poker();
+        //poker.winningPair(new String[]{"Q","Q"}, new String[]{"K","J"});
+        //poker.winningPair(new String[]{"Q","Q"}, new String[]{"K","K"});
+        poker.winningPairFromArray(new String[][]{new String[]{"9", "9"}, new String[]{"6", "6"}, new String[]{"7", "7"}, new String[]{"3", "3"}});
+
+    }
 }
```

Comment:
You've demonstrated a good understanding of the core task with your `winningPair` implementation, using a hashmap for card values effectively. Your logic in `winningPairFromArray` also shows a solid approach to determining the highest pair among multiple hands. And you've at least given ext 2 a think!